indico/core/db/sqlalchemy/locations.py

```diff
@@ -248,3 +248,48 @@ def location_data(self):
             return {'source': data_source, 'venue': data_source.venue, 'room': data_source.room,
                     'room_name': data_source.room_name, 'venue_name': data_source.venue_name,
                     'address': data_source.address, 'inheriting': self.inherit_location}
+
+    @property
+    def location_data_for_ui(self):
```

Comment:
I don't think we have time for that this release, and we'd still need some library like marshmallow that converts between the object and dumb structures (even if we'd just need it one-way). I think staying with util functions for now is the easiest way - adapting the code will be quick and easy once we have more time.